tools/link.py

```diff
@@ -1924,11 +1924,17 @@ def phase_emscript(in_wasm, wasm_target, js_syms, base_metadata):
   return metadata
 
 
-def run_embind_gen(wasm_target, js_syms, extra_settings):
+def run_embind_gen(wasm_target, js_syms, extra_settings, linker_inputs):
   # Save settings so they can be restored after TS generation.
   original_settings = settings.backup()
   settings.attrs.update(extra_settings)
 
+  if settings.MAIN_MODULE and linker_inputs:
+    # Copy libraries to the temp directory so they can be used when running
+    # in node.
+    for _i, linker_input in linker_inputs:
```

Comment:
Should we maybe filter by file name here so we don't copy object files? Looking at how the linker inputs work though, a side module can have any file ending.

Reply:
You can use `building.is_wasm_dylib()` I think